File: src/main/java/com/ridango/game/CocktailService.java
package com.ridango.game;

import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

@Service
public class CocktailService {

    private final java.lang.String API_URL = "https://www.thecocktaildb.com/api/json/v1/1/random.php";

    public Cocktail getRandomCocktail() {
        RestTemplate restTemplate = new RestTemplate();
        CocktailResponse response = restTemplate.getForObject(API_URL, CocktailResponse.class);
        return response.getDrinks().get(0);  // Get the first drink from the response
    }
}

File: src/services/cocktailService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/cocktail';

export const getRandomCocktail = () => {
    return axios.get(`${API_URL}/random`);
};

export const makeGuess = (guess) => {
    return axios.post(`${API_URL}/guess`, { guess });
};

export const getHint = () => {
    return axios.get(`${API_URL}/hint`);
};

export const startNewGame = () => {
    return axios.post(`${API_URL}/new-game`);
};
